apps/sms/views/shared/js/system_message_handler.js

```diff
@@ -270,11 +270,14 @@
 
         //Validate if message still exists before opening message thread
         //See issue https://bugzilla.mozilla.org/show_bug.cgi?id=837029
-        return Utils.onceDocumentIsVisible().then(
-          () => MessageManager.getMessage(id)
-        ).then(
+        return MessageManager.getMessage(id).then(
           (message) => Navigation.toPanel('thread', { id: message.threadId }),
```

Comment:
tell me what you think, I think we should even do this _before_ calling app.launch(). `toPanel().then(() => app.launch())`.